src/Backtracking.jl

```diff
@@ -0,0 +1,58 @@
+#=
+Backtracking line search implementation
+=#
+
+#=
+Input parameters:
+    - x: current iterate
+    - g: gradient 
+    - α: initial step size
+    - p: descent direction
+    - f: function evaluation
+=#
+
+mutable struct Backtracking{X, F, A}
+    x::X
+    g::X
+    p::X
+    f::F
+    α::A
+end
+
+# Determine step size by backtracking line search
+function Backtracking(backtrack::Backtracking)
```

Comment:
```suggestion
function Backtracking(;x, g,  α, p , f)
```

Reply:
Since this should be a constructor for the method

Reply:
I feel confused about this comment. In the function `Backtracking`, there are five arguments including `f`. Here `f` is a function. A function cannot be an argument of a function. That is where I feel confused.